R/snapshot.R

```diff
@@ -51,7 +51,9 @@ sd_snapshot <- function(self, private, items, filename, screenshot)
   content <- raw_to_utf8(req$content)
   content <- hash_snapshot_image_data(content)
   content <- jsonlite::prettify(content, indent = 2)
-  writeChar(content, file.path(current_dir, filename), eos = NULL)
+  con <- file(file.path(current_dir, filename), encoding = "UTF-8")
+  write(content, con)
+  close(con)
```

Comment:
It is possible for the `write` to throw an error, so the `close(con)` should be in an `on.exit()`, before `write()` is called.